src/timer.rs

```diff
@@ -0,0 +1,60 @@
+use std::{thread, time};
+use bus::Bus;
+use std::io::{self, Write};
+
+const SAVE: &str = "\x1b7";
+const RESTORE: &str = "\x1b8";
+const MOVE_TO_TOP_LEFT: &str = "\x1b[1;1H";
+const RED: &str = "\x1b[1;31m";
+const ERASE_LINE: &str = "\x1b[2K";
+const RESET: &str = "\x1b[0m";
+
+const TOTAL_TRANSACTION_TIME: u64 = 30;
+const REFRESH_INTERVAL: u64 = 100;
+
+pub struct Timer {
+    bus: Bus<String>
+}
+
+impl Timer {
+    pub fn new() -> Timer {
+        Timer {
+            bus: Bus::new(10)
+        }
+    }
+
+    pub fn start(&mut self) {
+        self.stop_timer();
+        self.run_timer();
+    }
+
+    pub fn stop(&mut self) {
+        self.stop_timer();        
+    }
+    
+    pub fn run_timer(&mut self) {
+        let mut recv = self.bus.add_rx();
+        thread::spawn(move || {
```

Comment:
I'm not so sure about the thread here. Is there a way to do this with async instead, such that we have concurrency but not parallelism? I think the way you wrote this is race-safe (given the move to top left + restore), but it'd probably make more sense to have this be part of the overall IO loop. This probably comes back to your question - can we do this with rustyline or switch to some other library that has animations/feedback as a feature?

Reply:
Currently, `rustyline` [does not provide an async interface](https://github.com/kkawakam/rustyline/issues/126). I'm currently looking into implementing this, however.

Reply:
Ideally we'd simply store the start time of the transaction in a variable somewhere and then define a render function that computed the remaining time based on `now-start`.

I'm not sure that the linked issue is relevant here. The missing piece is some sort of render hook.

Reply:
What should be responsible for the render function? Could you clarify on including this render process in to overall IO loop? Would the ideal solution not spawn any threads?